File: CSharpCodeBase/entities/ui/uiweaponcontroller.lua
local Class = require 'utils.hump.class'
local Entity = require 'unity.entity'
local RecipeList = require 'components.inventory.recipelist'
local UIGadgetController = require 'entities.ui.uigadgetcontroller'

local UIWeaponController= Class({__includes=Entity, init=function(self, gameObject)            
      Entity.init(self)   
      self.gameObject = gameObject
      self.weaponController = gameObject:GetComponent("UIWeaponController")
      self.weaponController:SetupLuaController(self)
      self.weaponController:SetCallback(self.Callback)     
      self.inputService = GameController.inputService
      self.activeWeaponSlot = 'weapon1'
end})

function UIWeaponController:GetCount()
  if GameController.inventory and GameController.gadgetController and #GameController.gadgetController.activeGadget > 0 then
    local count = GameController.inventory:GetItemCount(GameController.gadgetController.activeGadget[2] > 0)
    if count > 0 then
      return count
    else
    return ""
    end
  else
    return ""
  end
end

function UIWeaponController:SelectFirstWeapon()
  self.activeWeaponSlot = 'weapon1'
  local item = GameController.inventory.grids.equipment:GetItem(1)
  if item then
    GameController.player.playerController:GetCurrentSlot():Message("OnWeaponSelect", item.name)
  else
    GameController.player.playerController:GetCurrentSlot():Message("OnWeaponSelect", "hero_hand")
  end
end

function UIWeaponController:SelectSecondWeapon()
  self.activeWeaponSlot = 'weapon2'
  local item = GameController.inventory.grids.equipment:GetItem(2)
  if item then
    GameController.player.playerController:GetCurrentSlot():Message("OnWeaponSelect", item.name)
  else
    GameController.player.playerController:GetCurrentSlot():Message("OnWeaponSelect", "hero_hand")
  end
end

function UIWeaponController:SelectedSlotRepeated(weapon)
  if self.activeWeaponSlot == weapon then
    return true
  else
    return false
  end
end

function UIWeaponController:ActivateCurrentGadget()
  
end

return UIWeaponController

File: CSharpCodeBase/entities/ui/uiweaponcontroller.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class uiweaponcontroller :entity{
 var UIWeaponController= Class({__includes=Entity, init=function(self, gameObject)            
       Entity.init(self)   ;
       this.gameObject = gameObject;
       this.weaponController = gameObject:GetComponent("UIWeaponController");
       this.weaponController:SetupLuaController(self);
       this.weaponController:SetCallback(this.Callback)     ;
       this.inputService = GameController.inputService;
       this.activeWeaponSlot = "weapon1";
 }})
 public void GetCount(){
   if(GameController.inventory  &&  GameController.gadgetController  &&  #GameController.gadgetController.activeGadget > 0  ){ 
     var count = GameController.inventory:GetItemCount(GameController.gadgetController.activeGadget[2] > 0);
     if(count > 0  ){ 
       return count;
     }else{
     return "";
     }
   }else{
     return "";
   }
 }
 public void SelectFirstWeapon(){
   this.activeWeaponSlot = "weapon1";
   var item = GameController.inventory.grids.equipment:GetItem(1);
   if(item  ){ 
     GameController.player.playerController:GetCurrentSlot():Message("OnWeaponSelect", item.name);
   }else{
     GameController.player.playerController:GetCurrentSlot():Message("OnWeaponSelect", "hero_hand");
   }
 }
 public void SelectSecondWeapon(){
   this.activeWeaponSlot = "weapon2";
   var item = GameController.inventory.grids.equipment:GetItem(2);
   if(item  ){ 
     GameController.player.playerController:GetCurrentSlot():Message("OnWeaponSelect", item.name);
   }else{
     GameController.player.playerController:GetCurrentSlot():Message("OnWeaponSelect", "hero_hand");
   }
 }
 public void SelectedSlotRepeated(weapon){
   if(this.activeWeaponSlot == weapon  ){ 
     return true;
   }else{
     return false;
   }
 }
 public void ActivateCurrentGadget(){
   
 }
 }}